lib/api/bucketDeleteTagging.js

```diff
@@ -0,0 +1,52 @@
+const collectCorsHeaders = require('../utilities/collectCorsHeaders');
+const { metadataValidateBucket } = require('../metadata/metadataUtils');
+const { pushMetric } = require('../utapi/utilities');
+const metadata = require('../metadata/wrapper');
+const util = require('node:util');
+
+/**
+ * Bucket Delete Tagging - Delete a bucket's Tagging
+ * @param {AuthInfo} authInfo - Instance of AuthInfo class with requester's info
+ * @param {object} request - http request object
+ * @param {object} log - Werelogs logger
+ * @param {function} callback - callback to server
+ * @return {undefined}
+ */
+async function bucketDeleteTagging(authInfo, request, log, callback) {
+    const bucketName = request.bucketName;
+    let error = null;
+    log.debug('processing request', { method: 'bucketDeleteTagging', bucketName });
+
+    let bucket;
+    const metadataValidateBucketPromise = util.promisify(metadataValidateBucket);
+    let updateBucketPromise = util.promisify(metadata.updateBucket);
+    // necessary to bind metadata as updateBucket calls 'this', causing undefined otherwise
+    updateBucketPromise = updateBucketPromise.bind(metadata);
+    const metadataValParams = {
+        authInfo,
+        bucketName,
+        requestType: 'bucketDeleteTagging',
+    };
+
+    try {
+        bucket = await metadataValidateBucketPromise(metadataValParams, log);
+        bucket.setTags([]);
+        await updateBucketPromise(bucket.getName(), bucket, log);
+        pushMetric('deleteBucketTagging', log, {
+            authInfo,
+            bucket: bucketName,
+        });
+    } catch (err) {
+        error = err;
```

Comment:
for async functions I think we can just throw errors

Reply:
Related to my other comment, for now, the function flow is waiting for an error to be returned as part of the callback, if any. If we throw an error here, we will crash the project, is that what we want?